crypto-core/src/main/java/com/palantir/crypto2/io/ApacheCtrDecryptingSeekableInput.java

```diff
@@ -82,11 +82,13 @@ public int read(long position, byte[] buffer, int offset, int length) throws IOE
 
         @Override
         public int read(ByteBuffer dst) throws IOException {
-            byte[] bytes = new byte[dst.remaining()];
-            int read = input.read(bytes, 0, bytes.length);
+            if (readBuffer.length < dst.remaining()) {
+                resize(dst.remaining());
```

Comment:
Why is resizing necessary? Can't we read from the source in a loop using the buffer we already have?

Reply:
in practice the buffer should always fit, since we read into a byte buffer created with same size in https://github.com/palantir/hadoop-crypto/blob/e42e0496f6a1efe0a7c6a94d81509406672c5f5d/crypto-core/src/main/java/com/palantir/crypto2/io/ApacheCtrDecryptingSeekableInput.java#L45

so this covers the case that something changes in the calling code to use a larger buffer. looping intuitively feels more expensive, more os calls vs single allocation that gets reused from then on (assuming buffer size stays constant)

Reply:
Sure, I guess. The code gets more complicated but can do that